feat(environments): read project level notebook options

lorenzo-cavazzi opened the pull request:
When starting a new interactive environment, the UI now checks for project options available in the `.renku/renku.ini` file, it tries to apply the settings and it displays popup warnings where needed.
More info in the docs here: SwissDataScienceCenter/renku#723

**HOW TO TEST**
A preview is available at https://lorenzotest.dev.renku.ch/
This is a project with some already defined custom options: https://lorenzotest.dev.renku.ch/projects/lorenzo.cavazzi.tech/projectoptions

fix #481

![Screenshot_20200110_172044](https://user-images.githubusercontent.com/43481553/72168590-bcf97a00-33cd-11ea-8ef7-304f835dc822.png)

ciyer commented on src/notebooks/Notebooks.state.js:
```diff
@@ -205,21 +214,133 @@ class NotebooksCoordinator {
+        warnings: { $set: [] }
+      }
+    });
+    this.client.getRepositoryFile(projectId, RENKU_INI_PATH, filters.commit, 'raw')
```
Could the function argument to `getRepositoryFile` be broken into a few functions? I think that would help readability.

lorenzo-cavazzi replied:
Since most of the code was meant to convert the renku.ini raw content, I moved to its own function `NotebooksHelper.parseProjectOptions` with proper test coverage.
Also, the logic in the following `catch` function for handling missing files was somehow redundant. That case is now treated as an empty file

ciyer commented on src/notebooks/Notebooks.state.js:
```diff
@@ -205,21 +214,133 @@ class NotebooksCoordinator {
+    Object.keys(projectOptions).forEach(option => {
+      const optionValue = projectOptions[option];
+      if (
+        (globalOptions[option] && globalOptions[option].type !== "enum") ||
```
Could the inside of this `if` be extracted to a function? Something like `isOptionSupported`?

lorenzo-cavazzi replied:
Definitely. I moved those checks to its own function `NotebooksHelper.checkOptionValidity`, improved the logic to include type checking and added proper test coverage.

ciyer commented on src/utils/HelperFunctions.js:
```diff
@@ -63,5 +63,35 @@ function simpleHash(str) {
   return ("0000000" + (hval >>> 0).toString(16)).substr(-8);
 }
 
+function parseINIString(data) {
```
Could you write a few tests for this?

lorenzo-cavazzi replied:
Added

lorenzo-cavazzi commented:
I was a bit sloppy in reviewing the code before pushing.
I have now simplified the logic as suggested and added the corresponding tests
.
The preview on https://lorenzotest.dev.renku.ch/ has been updated.

ciyer commented:
Code looks good! I'm testing this feature, and I followed the documentation to request an amount of memory
```
* `mem_request [float]`: memory quota (in GBs)
```

and have created a renku.ini file
```
[renku "interactive"]
mem_request = 2
```
But it is not working as expected.

![image](https://user-images.githubusercontent.com/1196411/72431213-930cd280-3794-11ea-9e60-20a5b05c0fb9.png)

lorenzo-cavazzi commented:
Oh right, the `mem_request` field is not `2` but `2G`. I have to modify the docs to avoid confusion

lorenzo-cavazzi commented:
I have just pushed a fix for that SwissDataScienceCenter/renku#723
